tests/make-backup-and-restore-iso/runtest.sh

```diff
@@ -138,122 +107,14 @@ ISO_RECOVER_MODE=unattended' | tee $REAR_CONFIG" \
-                0 "Unmount $REAR_LABEL_PATH"
-        rlPhaseEnd
-
-        # RHEL 7 has EXTLINUX that does not support /boot being on XFS.  Use
-        # latest version from RHEL 8 instead.
-        if rlIsRHEL 7; then
-        rlPhaseStartSetup "[RHEL 7] Install newer EXTLINUX for chainloading"
-            rlRun "TmpDir=\"\$(mktemp -d)\"" 0 "Creating tmp directory"
-            rlRun "pushd '$TmpDir'" 0 "Change dir to $TmpDir"
-
-            # rlRpmDownload fails to download noarch packages :(
-            for pkg in syslinux syslinux-extlinux; do
-                rlRun -s "rlRpmDownload $pkg 6.04 5.el8 x86_64"
-                NO_ARCH_URL="$(grep -Po "(?<=trying download from ').*(?='$)" \
-                                        "$rlRun_LOG" | \
-                               sed 's/x86_64/noarch/g' | \
-                               sed "s/$pkg-6.04/$pkg-nonlinux-6.04/")"
-                rlRun "wget '$NO_ARCH_URL'" \
-                    0 "Download $pkg-nonlinux-6.04-5.el8.noarch.rpm"
-                rm "$rlRun_LOG"
-            done
-
-            rlRun "yum install -y ./syslinux*.rpm" 0 "Install EXTLINUX 6.04"
-
-            rlRun "popd '$TmpDir'" 0 "Change dir back"
-            rlRun "rm -rf $TmpDir" 0 "Removing tmp directory"
-
-            if ! rlGetPhaseState; then
-                rlDie "FATAL ERROR: failed to install EXTLINUX 6.04"
-            fi
-        rlPhaseEnd
-        fi
-
-        # Use EXTLINUX to chainload ReaR instead of GRUB as that did not work
-        # on some systems.
         rlPhaseStartSetup "Force the machine to autoboot the ReaR rescue system"
-            ROOT_DEVICE="$(lsblk -no pkname "$(df --output=source /boot | tail -n1)")"
-            KERNEL_VERSION="$(uname -r)"
-            SERIAL_DEVICE="$(grep -Eo '[0-9]+,.*' <<< "$CONSOLE_CMDLINE" \
-                             | tr ',' ' ')"
-
-            rlRun -l "extlinux --install /boot/extlinux" \
-                 0 "Install EXTLINUX to chainload ReaR"
-            rlRun -l "echo 'SERIAL $SERIAL_DEVICE
-UI menu.c32
-PROMPT 0
-
-MENU TITLE ReaR Chainload Boot Menu
-TIMEOUT 50
-
-LABEL linux
-    MENU LABEL $(grubby --default-title)
-    LINUX ../vmlinuz-$KERNEL_VERSION
-    APPEND $KERNEL_CMDLINE
-    INITRD ../initramfs-$KERNEL_VERSION.img
-
-LABEL rear
-    MENU LABEL Chainload ReaR from hd1
-    MENU DEFAULT
-    COM32 chain.c32
-    APPEND hd1' | tee /boot/extlinux/extlinux.conf" \
-                0 "Save EXTLINUX configuration"
-            rlRun "cat /usr/share/syslinux/mbr.bin > /dev/$ROOT_DEVICE" \
-                0 "Write EXTLINUX to /dev/$ROOT_DEVICE MBR"
-
-            if ! rlGetPhaseState; then
-                rlDie "FATAL ERROR: Installing EXTLINUX failed"
-            fi
+            rlLog "Setup GRUB"
+            rlRun "echo 'menuentry \"ReaR-recover\" {
+  loopback loop (hd0,msdos2)/root/rear/var/lib/rear/output/rear-fedora.iso
+  linux (loop)/isolinux/kernel rw selinux=0 console=ttyS0,9600 console=tty0 auto_recover unattended
```

Comment:
Also, is it intended that we boot the recovery directly from GRUB?  I know that it has some issues with chainloading but it would be nice to also test the bootloader (ISOLINUX AFAIK) stored on the ISO image.

Reply:
Ok, it looks like next commits have done exactly that...